### PR TITLE
DOC-713: New TableModified event

### PR DESCRIPTION
**Pull request by JamesToohey**
**Related Ticket:** 
DOC-713
TINY-6629

Description of Changes:
New `TableModified` event fired whenever a table is modified.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] ~~`_data/nav.yml` has been updated (if applicable)~~
- [x] ~~Files has been included where required (if applicable)~~
- [x] ~~Files removed have been deleted, not just excluded from the build (if applicable)~~
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: plugins/table.md [hunk 1]
```diff
@@ -696,12 +696,13 @@ tinymce.activeEditor.plugins.table.insertTable(2, 3);
...
-| newcell | Fired when a cell is created |
-| ObjectResizeStart | Fired when a resize action is started on a table, row, column or cell using the resize bars |
-| ObjectResized | Fired when a resize action is finished on a table, row, column or cell using the resize bars |
+| Name | Data | Description |
+|------| ---- | ----------- |
+| newrow | N/A | Fired when a row is created |
+| newcell | N/A | Fired when a cell is created |
+| ObjectResizeStart | N/A | Fired when a resize action is started on a table, row, column or cell using the resize bars |
+| ObjectResized | N/A | Fired when a resize action is finished on a table, row, column or cell using the resize bars |
+| tablemodified | `structure: bool; style: bool` | Fired whenever a table is modified in any way. The `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```
Comment (lnewson) [inline plugins/table.md:705]:
```suggestion
| TableModified | `{ structure?: boolean; style?: boolean }` | Fired whenever style or structural changes are made to a table. The `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```

**Review by lnewson — APPROVED** [review]

File: plugins/table.md (hunk 1 above)
Comment (lnewson) [reply plugins/table.md:705]: Note: I made `structure` and `style` optional in the above, as we aren't always passing them through atm.

Comment (JamesToohey) [reply plugins/table.md]: I added "The **optional** `style` and `structure` parameters..." too

File: plugins/table.md [hunk 2]
```diff
@@ -696,12 +696,13 @@ tinymce.activeEditor.plugins.table.insertTable(2, 3);
...
-| newcell | Fired when a cell is created |
-| ObjectResizeStart | Fired when a resize action is started on a table, row, column or cell using the resize bars |
-| ObjectResized | Fired when a resize action is finished on a table, row, column or cell using the resize bars |
+| Name | Data | Description |
+|------| ---- | ----------- |
+| newrow | N/A | Fired when a row is created |
+| newcell | N/A | Fired when a cell is created |
+| ObjectResizeStart | N/A | Fired when a resize action is started on a table, row, column or cell using the resize bars |
+| ObjectResized | N/A | Fired when a resize action is finished on a table, row, column or cell using the resize bars |
+| TableModified | `{ structure?: boolean, style?: boolean }` | Fired whenever style or structural changes are made to a table. The optional `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```
Comment (ltrouton) [inline plugins/table.md:705]: Just wanted to check/verify if we need to mention that `table` is also available or is that something we don't want to document?

File: release-notes/release-notes56.md [hunk 3]
```diff
@@ -44,6 +44,11 @@ A new optional `name` field that sets a specific name to the format when it's be
 
 For information on the `name` field, see: [User interface - `style_formats`]({{ site.baseurl }}/configure/editor-appearance/#style_formats).
 
+### New `TableModified` event
+
+The new `TableModified` event is fired whenever style or structural changes are made to a table.
+
+For more information on the `TableModified` event, see: [Table - `events`]({{ site.baseurl }}/plugins/table/#events).
```
Comment (ltrouton) [inline release-notes/release-notes56.md:51]: Minor thing and doesn't affect the output but I think there should be an extra line separating the text from the next heading like the others.

**Review by ltrouton — APPROVED** [review]

File: plugins/table.md [hunk 4]
```diff
@@ -696,12 +696,13 @@ tinymce.activeEditor.plugins.table.insertTable(2, 3);
...
-| newcell | Fired when a cell is created |
-| ObjectResizeStart | Fired when a resize action is started on a table, row, column or cell using the resize bars |
-| ObjectResized | Fired when a resize action is finished on a table, row, column or cell using the resize bars |
+| Name | Data | Description |
+|------| ---- | ----------- |
+| newrow | N/A | Fired when a row is created |
+| newcell | N/A | Fired when a cell is created |
+| ObjectResizeStart | N/A | Fired when a resize action is started on a table, row, column or cell using the resize bars |
+| ObjectResized | N/A | Fired when a resize action is finished on a table, row, column or cell using the resize bars |
+| TableModified | `{ structure?: boolean, style?: boolean }` | Fired whenever style or structural changes are made to a table (included in the event). The optional `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```
Comment (tylerkelly13) [inline plugins/table.md:705]:
```suggestion
| TableModified | `{ structure?: boolean, style?: boolean }` | Fired when style or structural changes are made to a table. The optional `style` and `structure` parameters describe whether the table style, structure, or both were modified. |
```

File: release-notes/release-notes56.md [hunk 5]
```diff
@@ -44,6 +44,12 @@ A new optional `name` field that sets a specific name to the format when it's be
 
 For information on the `name` field, see: [User interface - `style_formats`]({{ site.baseurl }}/configure/editor-appearance/#style_formats).
 
+### New `TableModified` event
+
+The new `TableModified` event is fired whenever style or structural changes are made to a table.
```
Comment (tylerkelly13) [inline release-notes/release-notes56.md:49]:
```suggestion
The new `TableModified` event is fired when style or structural changes are made to a table.
```

File: release-notes/release-notes56.md [hunk 6]
```diff
@@ -44,6 +44,12 @@ A new optional `name` field that sets a specific name to the format when it's be
...
+
+For more information on the `TableModified` event, see: [Table - `events`]({{ site.baseurl }}/plugins/table/#events).
```
Comment (tylerkelly13) [inline release-notes/release-notes56.md:51]:
```suggestion
For information on the `TableModified` event, see: [Table plugin - `events`]({{ site.baseurl }}/plugins/table/#events).
```

**Review by EkimChau — APPROVED** [review]

**Review by tylerkelly13 — APPROVED** [review]
:shipit:

**Pull request merged**